### PR TITLE
Add implementation of shift function

### LINKED ISSUE #1287: shift function implementation

### PR DESCRIPTION
**Pull request by prutskov**
## What do these changes do?

- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1287 and #1192? 
- [x] tests added and passing

### CODE REVIEW

**Comment by prutskov** [comment]
This changes are related with #1287

File: modin/pandas/test/test_dataframe.py [hunk 1]
```diff
@@ -2499,8 +2499,14 @@ def test_sem(self):
```
Comment (devin-petersohn) [inline modin/pandas/test/test_dataframe.py:2499]:
```suggestion

@pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
```

File: modin/pandas/test/test_dataframe.py [hunk 2]
```diff
@@ -2499,8 +2499,14 @@ def test_sem(self):
 
     def test_shift(self):
```
Comment (devin-petersohn) [inline modin/pandas/test/test_dataframe.py:2500]:
```suggestion
    def test_shift(self, data):
```

File: modin/pandas/test/test_dataframe.py [hunk 3]
```diff
@@ -2499,8 +2499,14 @@ def test_sem(self):
...
         data = test_data_values[0]
```
Comment (devin-petersohn) [inline modin/pandas/test/test_dataframe.py:2501]:
```suggestion
```

File: modin/pandas/test/test_series.py [hunk 4]
```diff
@@ -2231,9 +2231,10 @@ def test_shape(data):
 
 @pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
 def test_shift(data):
-    modin_series, _ = create_test_series(data)  # noqa: F841
-    with pytest.warns(UserWarning):
-        modin_series.shift()
+    modin_series, pandas_series = create_test_series(data)  # noqa: F841
```
Comment (devin-petersohn) [inline modin/pandas/test/test_series.py:2234]:
```suggestion
    modin_series, pandas_series = create_test_series(data)
```

**Comment by prutskov** [comment]
@devin-petersohn I implemented `shift` function by another way than using of `query_compiler`. Now this works only for `Series`. Is this approach you recommended me? If this is OK I will apply such approach for `DataFrame` case.

File: modin/pandas/base.py [hunk 5]
```diff
@@ -2551,10 +2551,78 @@ def set_axis(self, labels, axis=0, inplace=False):
...
+                    new_frame = dropped_df.append(filled_df, ignore_index=True)
+                    return new_frame
+            else:
+                if periods > 0:
+                    dropped_df = self.drop(self.columns[-periods:], axis="columns")
+                    new_frame = filled_df.join(dropped_df)
+                    return new_frame
+                else:
+                    dropped_df = self.drop(self.columns[:-periods], axis="columns")
+                    new_frame = dropped_df.join(filled_df)
```
Comment (devin-petersohn) [inline modin/pandas/base.py:2621]: I prefer `modin.pandas.concat` here.

File: modin/pandas/base.py [hunk 6]
```diff
@@ -2551,10 +2551,78 @@ def set_axis(self, labels, axis=0, inplace=False):
...
+                    new_frame = filled_df.append(dropped_df, ignore_index=True)
+                    return new_frame
+                else:
+                    dropped_df = self.drop(self.index[:-periods])
+                    new_frame = dropped_df.append(filled_df, ignore_index=True)
+                    return new_frame
+            else:
+                if periods > 0:
+                    dropped_df = self.drop(self.columns[-periods:], axis="columns")
+                    new_frame = filled_df.join(dropped_df)
```
Comment (devin-petersohn) [inline modin/pandas/base.py:2617]: I prefer `modin.pandas.concat` here.

File: modin/pandas/base.py (hunk 5 above)
Comment (prutskov) [reply modin/pandas/base.py]: fixed

File: modin/pandas/base.py (hunk 6 above)
Comment (prutskov) [reply modin/pandas/base.py]: fixed

**Pull request merged**